src/cache/cache.go

```diff
@@ -63,7 +64,7 @@ func (c *Cache) ProcessTxEntry(newTx *types.Transaction, currentTime int64) (Pro
 
 	utils.Logger.Debug().Msgf("Attempting to update cache with key [%s] and transaction hash [%s]", key, newTx.Hash().Hex())
 	if existing, found := c.Data[key]; found {
-		if existing.Tx != nil { // we sent a transaction in the last d seconds
+		if existing.TriedAgain { // we sent a transaction in the last d seconds
```

Comment:
Maybe to avoid any confusion with the previous internal retry mechanism, we could rename this flag to delayed, so we can understand if we are re-reading the code that the sending of the transaction was previously delayed, as it's not clear that it was externally retried by some wallet and not internally retried.